packages/rest-client/src/main/RestClient.ts

```diff
@@ -274,6 +308,23 @@ export default abstract class RestClient {
     })
   }
 
+  /**
+   * Make a request using underlying superagent instance, potentially getting a token.
+   *
+   * In an ideal world, this would rarely be used but provides an escape hatch to cater for bespoke usecases.
+   *
+   * @param call - A callback that provides the client a token and the underlying superagent instance.
+   * @param authOptions - The authentication options for this call.
+   * @returns The response body.
+   */
+  async makeRestClientCall<Response = unknown>(
```

Comment:
Gave this a bit of a clunky name which would be easy to github search for in the future